special-pages/pages/new-tab/app/omnibar/components/OmnibarProvider.js

```diff
@@ -0,0 +1,141 @@
+import { createContext, h } from 'preact';
+import { useCallback, useEffect, useReducer, useRef } from 'preact/hooks';
+import { useMessaging } from '../../types.js';
+import { reducer, useInitialDataAndConfig, useConfigSubscription } from '../../service.hooks.js';
+import { OmnibarService } from '../omnibar.service.js';
+
+/**
+ * @typedef {import('../../../types/new-tab.js').OmnibarConfig} OmnibarConfig
+ * @typedef {import('../../../types/new-tab.js').SuggestionsData} SuggestionsData
+ * @typedef {import('../../../types/new-tab.js').Suggestion} Suggestion
+ * @typedef {import('../../../types/new-tab.js').OpenTarget} OpenTarget
+ * @typedef {import('../../service.hooks.js').State<null, OmnibarConfig>} State // @todo: could this be never?
+ * @typedef {import('../../service.hooks.js').Events<null, OmnibarConfig>} Events // @todo: could this be never?
+ */
+
+/**
+ * These are the values exposed to consumers.
+ */
+export const OmnibarContext = createContext({
+    /** @type {State} */
+    state: { status: 'idle', data: null, config: null },
+    /** @type {(mode: OmnibarConfig['mode']) => void} */
+    setMode: () => {
+        throw new Error('must implement');
+    },
+    /** @type {(term: string) => Promise<SuggestionsData>} */
+    getSuggestions: () => {
+        throw new Error('must implement');
+    },
+    /** @type {(params: {suggestion: Suggestion, target: OpenTarget}) => void} */
+    openSuggestion: () => {
+        throw new Error('must implement');
+    },
+    /** @type {(params: {term: string, target: OpenTarget}) => void} */
+    submitSearch: () => {
+        throw new Error('must implement');
+    },
+    /** @type {(params: {chat: string, target: OpenTarget}) => void} */
+    submitChat: () => {
+        throw new Error('must implement');
+    },
+});
+
+export const OmnibarServiceContext = createContext(/** @type {OmnibarService|null} */ (null));
+
+/**
+ * A data provider that will use `OmnibarService` to fetch initial config only
+ *
+ * @param {Object} props
+ * @param {import("preact").ComponentChild} props.children
+ */
+export function OmnibarProvider(props) {
+    const initial = /** @type {State} */ ({
+        status: 'idle',
+        data: null,
+        config: null,
+    });
+
+    const [state, dispatch] = useReducer(reducer, initial);
+
+    // create an instance of `OmnibarService` for the lifespan of this component.
+    const service = useService();
+
+    // get initial data and config (data will be null)
+    useInitialDataAndConfig({ dispatch, service });
+
+    // subscribe to config updates
+    useConfigSubscription({ dispatch, service });
+
+    /** @type {(mode: OmnibarConfig['mode']) => void} */
+    const setMode = useCallback(
+        (mode) => {
+            service.current?.setMode(mode);
+        },
+        [service],
+    );
```

Comment:
Has it been decided for this to be a persisted value, rather than always defaulting back to search? Just curious since then you can handle it on the FE without the wiring.

Can always change before next PR so don't worry about changing it now if you don't know

Reply:
I’ll raise this in https://app.asana.com/1/137249556945/project/1209121419454298/task/1210610383905028?focus=true as the short answer is I am undecided. Easy to change later, as you say.